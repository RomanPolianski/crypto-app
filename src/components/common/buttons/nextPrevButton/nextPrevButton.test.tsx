import { fireEvent, render, screen } from '../../../../../test-utils';
import { NextPrevButton } from './NextPrevButton';

const onclick = () => {};

describe('Close btn test', () => {
  it('is round delete visible & disabled', () => {
    const NextPrevbtn = render(
      <NextPrevButton
        color="black"
        type="prev"
        disabled={false}
        onclick={onclick}
      />
    );
    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(screen.getByRole('button')).not.toBeDisabled();
  });

  it('is button square close visible', () => {
    const NextPrevbtn = render(
      <NextPrevButton color="white" type="prev" disabled onclick={onclick} />
    );
    expect(screen.getByRole('button')).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeDisabled();
  });

  it('is button clickable', () => {
    const NextPrevbtn = render(
      <NextPrevButton color="white" type="prev" disabled onclick={onclick} />
    );

    const btn = screen.getByRole('button');
    fireEvent.click(btn);
  });

  it('Snapshot button round delete', () => {
    const NextPrevbtn = render(
      <NextPrevButton color="black" type="prev" disabled onclick={onclick} />
    );
    expect(NextPrevbtn).toMatchSnapshot();
  });

  it('Snapshot button square close', () => {
    const NextPrevbtn = render(
      <NextPrevButton color="white" type="prev" disabled onclick={onclick} />
    );
    expect(NextPrevbtn).toMatchSnapshot();
  });
});
