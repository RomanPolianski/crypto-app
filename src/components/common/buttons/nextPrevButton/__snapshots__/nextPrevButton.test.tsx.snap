// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close btn test > Snapshot button round delete 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="_nextprevButton_18cvp_1"
        disabled=""
        type="button"
      >
        <span>
          <svg
            class="bi bi-arrow-left-circle"
            fill="currentColor"
            height="30"
            viewBox="0 0 16 16"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Prev
            </title>
             
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
              fill-rule="evenodd"
            />
             
          </svg>
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="_nextprevButton_18cvp_1"
      disabled=""
      type="button"
    >
      <span>
        <svg
          class="bi bi-arrow-left-circle"
          fill="currentColor"
          height="30"
          viewBox="0 0 16 16"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Prev
          </title>
           
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
            fill-rule="evenodd"
          />
           
        </svg>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Close btn test > Snapshot button square close 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="_nextprevButton_18cvp_1 _nextprevButton__white_18cvp_22"
        disabled=""
        type="button"
      >
        <span>
          <svg
            class="bi bi-arrow-left-circle"
            fill="currentColor"
            height="30"
            viewBox="0 0 16 16"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Prev
            </title>
             
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
              fill-rule="evenodd"
            />
             
          </svg>
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="_nextprevButton_18cvp_1 _nextprevButton__white_18cvp_22"
      disabled=""
      type="button"
    >
      <span>
        <svg
          class="bi bi-arrow-left-circle"
          fill="currentColor"
          height="30"
          viewBox="0 0 16 16"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Prev
          </title>
           
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
            fill-rule="evenodd"
          />
           
        </svg>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Close btn test Snapshot button round delete 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="nextprevButton"
        disabled=""
        type="button"
      >
        <span
          class="svg"
        >
          <svg
            class="bi bi-arrow-left-circle"
            fill="currentColor"
            height="30"
            viewBox="0 0 16 16"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Prev
            </title>
             
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
              fill-rule="evenodd"
            />
             
          </svg>
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="nextprevButton"
      disabled=""
      type="button"
    >
      <span
        class="svg"
      >
        <svg
          class="bi bi-arrow-left-circle"
          fill="currentColor"
          height="30"
          viewBox="0 0 16 16"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Prev
          </title>
           
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
            fill-rule="evenodd"
          />
           
        </svg>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Close btn test Snapshot button square close 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="nextprevButton nextprevButton__white"
        disabled=""
        type="button"
      >
        <span
          class="svg"
        >
          <svg
            class="bi bi-arrow-left-circle"
            fill="currentColor"
            height="30"
            viewBox="0 0 16 16"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Prev
            </title>
             
            <path
              d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
              fill-rule="evenodd"
            />
             
          </svg>
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="nextprevButton nextprevButton__white"
      disabled=""
      type="button"
    >
      <span
        class="svg"
      >
        <svg
          class="bi bi-arrow-left-circle"
          fill="currentColor"
          height="30"
          viewBox="0 0 16 16"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Prev
          </title>
           
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zm-4.5-.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
            fill-rule="evenodd"
          />
           
        </svg>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
